product_multi_ean/models/product_product.py

```diff
@@ -28,18 +28,6 @@ class ProductEan13(models.Model):
         required=True,
     )
 
-    @api.multi
-    @api.constrains('name')
-    @api.onchange('name')
-    def _check_name(self):
-        barcode_obj = self.env['barcode.nomenclature']
-        for record in self.filtered('name'):
-                if not barcode_obj.check_ean(record.name):
```

Comment:
In fact, the function called in this check was misused.

You cannot just remove this but change the logic.

You have to call this :

https://github.com/odoo/odoo/blob/11.0/addons/barcodes/models/barcodes.py#L125

And then, call :

https://github.com/odoo/odoo/blob/11.0/addons/barcodes/models/barcodes.py#L57

with the first provided encoding.

If you want to implement a new one (GS1), simply override these functions

Reply:
By default, Odoo allow add any barcode to product barcode field, ¿Why this module add this restriction?

Reply:
I Know that this restrictions are used when a barcode is readed... but not when a product is created

Reply:
You have to ask the author :smile: 

See refactoring commit :

https://github.com/OCA/stock-logistics-barcode/commit/3fa99ac3a558d1913f72fbe5549346fd0fd3d8cb#diff-71824b7fc2c784066fee1144d32b283a

@pedrobaeza

Reply:
That made sense when done on previous version, but not now. It's something that should be taken into account when migrating